### ISSUE TITLE
Wrong CalendarValidatorFactory being used?

### ISSUE THREAD

**Issue by brianlenz** [ical4j/ical4j-extensions#9]
We are using iCal4J 2.0.0 with iCal4J extensions 1.0.0 (all the latest). When constructing a new Calendar object, the AbstractCalendarValidatorFactory is setting its static CalendarValidatorFactory instance to a DefaultCalendarValidatorFactory instead of the expected ExtensionsCalendarValidatorFactory, which happens to be second in the Iterator returned by ServiceLoader. Is this a bug or is there something that should be done to get the classes to load in the proper order so that the ExtensionsCalendarValidatorFactory is used? As it stands right now, we are required to use the Calendar constructor that takes the Validator in order to explicitly set the ExtensionsCalendarValidator as the validator to use. Otherwise, any Calendars that use extensions will fail validation.

FWIW, we were using iCal4J 1.0.6 and iCal4J extensions 0.9.2 previously and didn't need to do anything special to handle the extensions validation (though I see the validation framework was handled completely differently on those older versions).

Thanks for providing a great open source library!

**Comment by benfortuna** [comment]
Yes it does appear to be a problem with how service loader works.. 

I think we'll need to fix this (for now) by loading and applying all calendar validators available. I've started work on this on ical4j branch: bugfix/validator-serviceloader

regards,
ben

**Comment by brianlenz** [comment]
Awesome thank you!
